addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -286,6 +295,10 @@ function factory(dependencies) {
 
     ThreadCache.fields = {
         checkedMessages: many2many('mail.message', {
+            compute: '_computeCheckedMessages',
+            dependencies: [
+                'messagesCheckboxes',
+            ],
```

Comment:
Forgot about it but it should have itself has dependency too, to ensure the compute is also done when adding new values, to prevent adding bad values in the first place.